### ISSUE TITLE
Rule Request: delete unused auto-generated code

### ISSUE THREAD

**Issue by yonat** [realm/SwiftLint#2578]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

> 1. Why should this rule be added?

Xcode auto-generates function like `didReceiveMemoryWarning()` that are often unused, and just left in code to rot.

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Triggering:

```swift
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func applicationWillResignActive(_ application: UIApplication) {
        // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
        // Use this method to pause ongoing tasks, disable timers, and invalidate graphics rendering callbacks. Games should use this method to pause the game.
    }

    override func viewDidDisappear(_ animated: Bool) {
        super.viewDidDisappear(animated)
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
```

Non-triggering:

```swift
func emptyImplementationForRquiredProtocolFunction() {
    }
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

No params, I think.
> 4. Should the rule be opt-in or enabled by default? Why?

Enabled by default, unused code should be deleted.

**Comment by jpsim** [comment]
I would welcome a rule that validated that overridden methods do more than simply call `super`.
